alg/teca_temporal_reduction.cxx

```diff
@@ -2355,21 +2369,78 @@ const_p_teca_dataset teca_temporal_reduction::execute(
 
     // copy the first mesh
-    const_p_teca_cartesian_mesh mesh_in
-      = std::dynamic_pointer_cast<const teca_cartesian_mesh>(data_in[n_data-1]);
+    p_teca_cartesian_mesh mesh_in
+      = std::dynamic_pointer_cast<teca_cartesian_mesh>(
+      std::const_pointer_cast<teca_dataset>(data_in[n_data-1]));
     p_teca_cartesian_mesh mesh_out = teca_cartesian_mesh::New();
     mesh_out->shallow_copy(
       std::const_pointer_cast<teca_cartesian_mesh>(mesh_in));
     p_teca_array_collection arrays_out = mesh_out->get_point_arrays();
 
     size_t n_array = this->point_arrays.size();
 
+    bool firstIter = true;
+
     // accumulate incoming values
-    for (int i = n_data-2; i >= 0; --i)
+    for (int i = n_data-2; i >= 0 || firstIter; --i)
     {
-        mesh_in
-           = std::dynamic_pointer_cast<const teca_cartesian_mesh>(data_in[i]);
-        const_p_teca_array_collection arrays_in = mesh_in->get_point_arrays();
+        firstIter = false;
+
+        if (n_data >= 2)
+        {
+            mesh_in = std::dynamic_pointer_cast<teca_cartesian_mesh>(
+              std::const_pointer_cast<teca_dataset>(data_in[i]));
+        }
+
+        p_teca_array_collection arrays_in = mesh_in->get_point_arrays();
+
+        // Handle the case where the number of inputs < 2.
+        // The arrays_in is to all zero or the fill_value if one is provided.
+        if (n_data < 2)
+        {
+            for (size_t j = 0; j < n_array; ++j)
+            {
+                const std::string &array = this->point_arrays[j];
+
+                const_p_teca_variant_array in_array = arrays_in->get(array);
+                unsigned long n_elem = in_array->size();
+
+#if defined(TECA_HAS_CUDA)
+                if (device_id >= 0)
+                {
+                    p_teca_variant_array red_array = in_array->new_instance(n_elem, teca_variant_array::allocator::cuda);
+                }
+                else
+                {
+#endif
+                    p_teca_variant_array red_array = in_array->new_instance(n_elem);
+#if defined(TECA_HAS_CUDA)
+                }
+#endif
+                TEMPLATE_DISPATCH(
+                    teca_variant_array_impl,
+                    red_array.get(),
+
+#if defined(TECA_HAS_CUDA)
+                    if (device_id >= 0)
+                    {
+                        auto sp_red_array = static_cast<TT*>(red_array.get())->get_cuda_accessible();
+                    }
+                    else
+                    {
+#endif
+                        auto sp_red_array = static_cast<TT*>(red_array.get())->get_cpu_accessible();
+#if defined(TECA_HAS_CUDA)
+                    }
+#endif
+                    NT *p_red_array = sp_red_array.get();
+
+                    for (unsigned int i = 0; i < n_elem; ++i)
+                        p_red_array[i] = (this->fill_value != -1) ? this->fill_value : 0.;
+                )
+                arrays_in->set(array, red_array);
+            }
+        }
```

Comment:
handle the n_data < 2 case outside the for loop.

minor: n_data has 64 bits, better to use a 64 bit integer in the loop counter.

Reply:
Do you mean `unsigned long` instead of `size_t`?

Reply:
nope, I mean use long (64 bit) instead of int (32 bit) in the loop count because n_data is size_t (64 bit). when mixing 32 and 64 bit types there can be an overflow on the 32 bit type.

Reply:
also, I suspect there will be a compiler warning generated for this code because an overflow can also happen when mixing signed and unsigned types. Because of the intial loop count (i = n_data - 2) could be negative in some case it's probably better to use a signed data type for n_data such as ssize_t or long as well.